src/components/Login/SignUp.js

```diff
@@ -0,0 +1,168 @@
+import React, { Component } from 'react';
+import {
+ Alert,
+ Image,
+ Text,
+ TextInput,
+ TouchableWithoutFeedback,
+ TouchableHighlight,
+ View,
+} from 'react-native';
+import { Actions } from 'react-native-router-flux';
+import ErrorMeta from '../../utils/ErrorMeta';
+import LinearGradient from 'react-native-linear-gradient';
+import LoginUtil from '../../utils/LoginUtil';
+import ServerUtil from '../../utils/ServerUtil';
+import styles from './Styles';
+
+class Login extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      email: '',
+      password1: '',
+      password2: '',
+    };
+
+    LoginUtil.initCallback(this.onLoginSuccess, this.onServerFail);
+  }
+
+  onLoginSuccess(result) {
+    if (result === undefined) {
+      Actions.login();
+    } else {
+      Actions.generalInfo();
+    }
+  }
+
+  onServerFail(error) {
+    if (error.code !== ErrorMeta.ERR_NONE) {
+      alert(error.msg);
+    }
+  }
+
+  render() {
+    let onChangeEmail = (text) => { this.state.email = text; };
+    let onChangePassword1 = (text) => { this.state.password1 = text; };
+    let onChangePassword2 = (text) => { this.state.password2 = text; };
+    let createAccount = () => this.createAccount();
+
+    return (
+
+      //  Render the screen on View.
+      <View style={styles.container}>
+        <View style={styles.mainLogo}>
+          <Image source={require('../../resources/splash_icon_1x.png')} />
+        </View>
+
+        {/* Render facebook login button */}
+        <TouchableWithoutFeedback onPress={LoginUtil.signInWithFacebook}>
+          <View style={[styles.facebookLoginContainer, { marginTop: 43 }]}>
+            <Image style={styles.facebookLoginButton}
+                   source={require('../../resources/fb.png')} />
+            <Text style={styles.facebookLoginText}>Login with Facebook</Text>
+          </View>
+        </TouchableWithoutFeedback>
+
+        <View style={[styles.hrContainer, { marginTop: 10 }]}>
+          <View style={styles.hr}></View>
+          <View><Text style={styles.hrText}>or</Text></View>
+          <View style={styles.hr}></View>
+        </View>
+
```

Comment:
As you know, legibility is important factor, so I think empty line is necessary in some case. But, If we have to follow guideline, Okay! I'll